app/controllers/system_identity_permissions_controller.rb

```diff
@@ -97,11 +112,23 @@ def update
       type: 'system',
       group_id: @group_id,
       successes: successes,
-      fails: fails
+      fails: fails,
+      overwrite_fails: overwrite_fails,
+      revision_ids: next_revision_ids
     )
 
-    flash[:success] = 'System Object Permissions were saved.' unless successes.blank?
-    flash[:error] = "#{fails.join(', ')} permissions were unable to be saved." unless fails.blank?
+    unless successes.blank?
+      flash[:success] = successes.join(', ').titleize + ' permissions were saved.'
+    end
+    unless fails.blank? && overwrite_fails.blank?
```

Comment:
I think this `unless` block is a little more convoluted than necessary as is.
Line 123 can be removed as 124 and 125 are already doing the `unless` check.
For the rest, this is just a suggestion, but to make it more readable, I might try:
```
if error_message.present? && overwrite_error_message.present?
  ...
elsif error_message.present?
  ...
elsif overwrite_error_message.present?
  ...
end
```